src/opensearch_sdk_py/settings/settings.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+
+
+from typing import Any, Dict, Optional, Union
+
+from opensearch_sdk_py.transport.stream_input import StreamInput
+from opensearch_sdk_py.transport.stream_output import StreamOutput
+
+
+class Settings:
```

Comment:
Are settings typed? If so this interface should only have `get` and return the right type of value, the caller shouldn't know about what to ask for.

Reply:
> Are settings typed? If so this interface should only have `get` and return the right type of value, the caller shouldn't know about what to ask for.

I overcomplicated my previous reply to this.  Here's the TLDR:
1. The `Settings` class is basically a holder for a `Map<String, Object>` with setting name to its value (one of many possible types including user-defined).
2. The `Setting<T>` class (not yet implemented here) contains all the logic for the setting including parsing, validating, min/max, default values.
3. The `get()` on the `Settings` class returns a `String` that is parsed by its corresponding `Setting`.  Basically anytime you want the value of a `Setting` you pass it the current system `Settings` map, and it looks it up and responds appropriately.

So the `getAsX()` methods will probably not ever be used on Extensions and can be deleted, we'll use `Setting.get(Settings)` to get the appropriate return value.